### PR TITLE
Update tekton tasks manifests to version v0.10.0 

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
Update tekton tasks manifests to version v0.10.0
Update necessary code to onboard new create-data-object task

**Release note**:
```
Onboard create-data-object-task
```

### CODE REVIEW

File: data/olm-catalog/tekton-tasks-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -261,7 +270,7 @@ spec:
                   valueFrom:
                     fieldRef:
                       fieldPath: metadata.namespace
-                image: quay.io/kubevirt/tekton-tasks-operator:v0.1.2
+                image: quay.io/ksimon/tekton-tasks-operator:latest
```
Comment (orenc1) [inline data/olm-catalog/tekton-tasks-operator.clusterserviceversion.yaml:273]: Hi @ksimon1 ,
Is this change intended, using your own quay account instead of kubevirt's ?

Comment (0xFelix) [reply data/olm-catalog/tekton-tasks-operator.clusterserviceversion.yaml:273]: This was very likely not intended but it is also overwritten by the csv-generator during release. We will however still revert it.
